ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/clientlibs/site/js/textinputview.js

```diff
@@ -120,6 +122,46 @@
 
         _updateValue(value) {
             this.widget.value = value;
+            this._runValidations(this.widget);
+        }
+
+        _runValidations(widget) {
+            const model = this._model._jsonModel;
+            let validationObj = {isError: false, errorMessage: ''};
+            this._runRequiredValidation(widget, model, validationObj);
+            this._runMinMaxValidation(widget, model, validationObj);
+            this._toggleErrorMark(validationObj);
+        }
+
+        _runRequiredValidation(widget, model, validationObj) {
+            if (model.required && widget.value === '') {
+                validationObj.isError = true;
+                validationObj.errorMessage = model.constraintMessages['required'];
+            }
+        }
+
+        _runMinMaxValidation(widget, model, validationObj) {
```

Comment:
this should be done by model, not view. view should just act on model reporting some invalidation in subscribe API.

Reply:
Noted. Thanks for such a quick review! :) It helps a lot.